test/groovy/MTABuildTest.groovy

```diff
@@ -171,6 +171,41 @@ public class MTABuildTest extends BasePipelineTest {
     }
 
 
+    @Test
+    void mtaJarLocationFromCustomStepConfigurationTest(){
+
+        setupCommonPipelineEnvironmentScript.call(script: [commonPipelineEnvironment: cpe])
+        mtaBuildScript.call(script: [commonPipelineEnvironment: cpe],
+                      buildTarget: 'NEO')
+
+        assert jscr.shell[1].contains('-jar /custom/mta/mta.jar --mtar')
+        assert jlr.log.contains("[mtaBuild] MTA JAR '/custom/mta/mta.jar' retrieved from configuration.")
+    }
+
+
+    @Test
+    void mtaJarLocationFromDefaultStepConfigurationTest() {
+
+        helper.registerAllowedMethod("readYaml", [Map], { Map parameters ->
+            if(parameters.text) {
+                return new Yaml().load(parameters.text)
+            } else if(parameters.file) {
+                if (parameters.file == '.pipeline/config.yml') return [:]
+                return new Yaml().load(new File(parameters.file).text)
+            } else {
+                throw new IllegalArgumentException("Key 'text' is missing in parameters $parameters.")
+            }
+        })
+
+        setupCommonPipelineEnvironmentScript.call(script: [commonPipelineEnvironment: cpe])
```

Comment:
you might be able to get completely rid of `setupCommonPipelineEnvironment`-> see below

Reply:
But why would I want to get rid of it? Should I not mock the original flow? that as you said is like this:

"IMHO we should rely on setupCommonPipelineEnvironment being called before.

Typical flow in a pipeline would be:

```
setUpCommonPipelineEnvironment ...
artifactSetVersion ...
mtaBuild ...
...
```"

Reply:
For a integration-like test I agree.
For a unit-like test it should be sufficient to just test the logic in the step.
This is what we do in the tests for the other steps.

Reply:
Ok. I do it once more. At least I did learn the original flow :)